Week1/LongestSubString.java

```diff
@@ -0,0 +1,40 @@
+import java.util.HashSet;
+
+public class LongestSubString {
+    public static int subString(String input){
+
+    	/* Invalid Input testing */
+    	if (input.contains(" "))
+    	{
+    		return 0; //Zero Length if it is empty string or contains space.
+    	}
+    		
+    	String LongestStr = "";
+        String DiscoveredLongest = "";
+        HashSet<Character> set = new HashSet<Character>(); //for no repetition in character set
+
+        
+
+        for (int i = 0; i < input.length(); i++) 
+        {
+            char c = input.charAt(i); //Charactered Appered in scanning of string
+
+            if (set.contains(c)) //if character already in scanned character set
+            {
+            	DiscoveredLongest = ""; //discard probable longest string
+                set.clear();			//Refresh Stored Character set
+            }
+            DiscoveredLongest += c;    //Otherwise concatenate appeared character in substring

```

Comment:
Using += creates a whole new string in memory because Java strings are immutable, so this can be inefficient

Think about what you care about here; you never actually use the values of DiscoveredLongest or LongestStr, only their Length. Why store the string if you don't have to, maybe you can store a different, non-immutable value instead :)